erpnext/utilities/bulk_transaction.py

```diff
@@ -0,0 +1,128 @@
+import json
+
+import frappe
+
+
+@frappe.whitelist()
+def transaction_processing(data, to_create):
+	deserialized_data = json.loads(data)
+	length_of_data = len(deserialized_data)
+
+	if length_of_data > 10:
+		# frappe.msgprint("Started a background job to create {1} {0}".format(to_create,length_of_data))
+		frappe.enqueue(job, deserialized_data=deserialized_data, to_create=to_create)
+	else:
+		job(deserialized_data, to_create)
+
+def job(deserialized_data, to_create):
+	from erpnext.accounts.doctype.payment_entry import payment_entry
+	from erpnext.accounts.doctype.purchase_invoice import purchase_invoice
+	from erpnext.accounts.doctype.sales_invoice import sales_invoice
+	from erpnext.buying.doctype.purchase_order import purchase_order
+	from erpnext.buying.doctype.supplier_quotation import supplier_quotation
+	from erpnext.selling.doctype.quotation import quotation
+	from erpnext.selling.doctype.sales_order import sales_order
+	from erpnext.stock.doctype.delivery_note import delivery_note
+	from erpnext.stock.doctype.purchase_receipt import purchase_receipt
+
+	i = 0
+	for d in deserialized_data:
+		try:
+			i+=1
+
+			# From Sales Order
+			if to_create == "Sales Invoice From Sales Order":
```

Comment:
Too much code duplication here?

Python allows treating functions as first class objects, just create a mapping like this instead: 

If a particular function needs non-standard argument you create a partial function: https://docs.python.org/3/library/functools.html#functools.partial 


`Tuple[from_doctype, to_doctype] -> Function`